### PR TITLE
Adding BoxFit.cover to CircleAvatar

### LINKED ISSUE #13478: CircleAvatar needs a fit property
CircleAvatar needs a **fit** property

```
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
import 'package:flutter/material.dart';

// Examples can assume:
// String userAvatarUrl;

/// A circle that represents a user.
///
/// Typically used with a user's profile image, or, in the absence of
/// such an image, the user's initials. A given user's initials should
/// always be paired with the same background color, for consistency.
///
/// ## Sample code
///
/// If the avatar is to have an image, the image should be specified in the
/// [backgroundImage] property:
///
/// ```dart
/// new CircleAvatar(
///   backgroundImage: new NetworkImage(userAvatarUrl),
/// )
/// ```
///
/// The image will be cropped to have a circle shape.
///
/// If the avatar is to just have the user's initials, they are typically
/// provided using a [Text] widget as the [child] and a [backgroundColor]:
///
/// ```dart
/// new CircleAvatar(
///   backgroundColor: Colors.brown.shade800,
///   child: new Text('AH'),
/// )
/// ```
///
/// See also:
///
///  * [Chip], for representing users or concepts in long form.
///  * [ListTile], which can combine an icon (such as a [CircleAvatar]) with some
///    text for a fixed height list entry.
///  * <https://material.google.com/components/chips.html#chips-contact-chips>
class CircleAvatar2 extends StatelessWidget {
  /// Creates a circle that represents a user.
  const CircleAvatar2({
    Key key,
    this.child,
    this.backgroundColor,
    this.backgroundImage,
    this.foregroundColor,
    this.radius: 20.0,
    this.fit: BoxFit.cover,
  })
      : super(key: key);

  /// The widget below this widget in the tree.
  ///
  /// Typically a [Text] widget. If the [CircleAvatar] is to have an image, use
  /// [backgroundImage] instead.
  final Widget child;

  /// The color with which to fill the circle. Changing the background
  /// color will cause the avatar to animate to the new color.
  ///
  /// If a background color is not specified, the theme's primary color is used.
  final Color backgroundColor;

  /// The default text color for text in the circle.
  ///
  /// Falls back to white if a background color is specified, or the primary
  /// text theme color otherwise.
  final Color foregroundColor;

  /// The background image of the circle. Changing the background
  /// image will cause the avatar to animate to the new image.
  ///
  /// If the [CircleAvatar] is to have the user's initials, use [child] instead.
  final ImageProvider backgroundImage;

  /// The size of the avatar. Changing the radius will cause the
  /// avatar to animate to the new size.
  ///
  /// Defaults to 20 logical pixels.
  final double radius;

  /// Avatar Image BoxFit.
  ///
  /// Defaults to cover.
  final BoxFit fit;

  @override
  Widget build(BuildContext context) {
    assert(debugCheckHasMediaQuery(context));
    final ThemeData theme = Theme.of(context);
    TextStyle textStyle = theme.primaryTextTheme.title;
    if (foregroundColor != null) {
      textStyle = textStyle.copyWith(color: foregroundColor);
    } else if (backgroundColor != null) {
      switch (ThemeData.estimateBrightnessForColor(backgroundColor)) {
        case Brightness.dark:
          textStyle = textStyle.copyWith(color: Colors.white);
          break;
        case Brightness.light:
          textStyle = textStyle.copyWith(color: Colors.black);
          break;
      }
    }
    return new AnimatedContainer(
      width: radius * 2.0,
      height: radius * 2.0,
      duration: kThemeChangeDuration,
      decoration: new BoxDecoration(
        color: backgroundColor ?? theme.primaryColor,
        image: backgroundImage != null
            ? new DecorationImage(
                image: backgroundImage,
                fit: fit,
              )
            : null,
        shape: BoxShape.circle,
      ),
      child: child != null
          ? new Center(
              child: new MediaQuery(
              // Need to reset the textScaleFactor here so that the
              // text doesn't escape the avatar when the textScaleFactor is large.
              data: MediaQuery.of(context).copyWith(textScaleFactor: 1.0),
              child: new DefaultTextStyle(
                style: textStyle.copyWith(color: foregroundColor),
                child: child,
              ),
            ))
          : null,
    );
  }
}

```

### PR DESCRIPTION
**Pull request by harrisonturton**
Currently the `CircleAvatar` does not have a `fit` property. This causes rectangular images to shrink not fully cover the circle. This looks pretty bad.
I've set it to a constant `BoxFit.cover`, because the image should always cover the whole avatar.

Fixes #13478.

This is my first contribution! Let me know if I should've done anything differently.

### CODE REVIEW

**Comment by harrisonturton** [comment]
I signed it!

File: bin/internal/engine.version [hunk 1]
```diff
@@ -1 +1 @@
-ac4089145203a19d317747c32c9114dd4894aa82
+23c0b7bbf77762d774cb96f93af0016776ec60c4
```
Comment (Hixie) [inline bin/internal/engine.version]: was this intentional?

File: packages/flutter/lib/src/material/circle_avatar.dart [hunk 2]
```diff
@@ -179,7 +179,14 @@ class CircleAvatar extends StatelessWidget {
...
         color: effectiveBackgroundColor,
-        image: backgroundImage != null ? new DecorationImage(image: backgroundImage) : null,
+        image: backgroundImage != null
+			? new DecorationImage(
+				image: backgroundImage,
+				// Otherwise rectangular images are shrunk down
+				// and do not cover the entire avatar.
+				fit: BoxFit.cover
+			),
+			: null,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/circle_avatar.dart]: Let's indent this slightly differently so it's more consistent with the rest of the code, e.g.:

```dart
        image: backgroundImage == null ? null :
          new DecorationImage(
            image: backgroundImage,
            fit: BoxFit.cover,
          ),
```

**Comment by Hixie** [comment]
This will need a test.

I'm not really sure what is making Travis angry... maybe the tabs?

**Comment by Hixie** [comment]
@harrisonturton Will you have the opportunity to integrate the above review comments? It's totally fine if you don't have time to do this, we'd be happy to (eventually) take care of it. Thanks for your contribution!

**Comment by gspencergoog** [comment]
This PR has been replaced by https://github.com/flutter/flutter/pull/16649

**Pull request closed without merging**